test/components/remote_retro_test.js

```diff
@@ -1,7 +1,7 @@
 import React from "react"
 import { spy } from "sinon"
 
-import { RemoteRetro } from "../../web/static/js/components/remote_retro"
+import { RemoteRetro, isNewFacilitator, findCurrentUser, findFacilitatorName } from "../../web/static/js/components/remote_retro"
```

Comment:
There's a really nice opportunity to extract the `findCurrentUser`  and `findFacilitatorName` functions out into redux selectors, as we're already [determining the facilitator's name within the `LobbyStage` component](https://github.com/stride-nyc/remote_retro/blob/5515c41d36e68c19e49552cc99c5cd8102a7173c/web/static/js/components/lobby_stage.jsx#L12-L13), and by leveraging these selectors in the `mapStateToProps` call in RemoteRetro, we can greatly reduce the amount of logic that lives (or needs to be invoked) within `RemoteRetro#componentDidUpdate`. Allows us to DRY up the codebase a bit!

Reply:
A nice read [outlining and explaining the benefits of the redux selector pattern](https://hackernoon.com/selector-pattern-painless-redux-store-destructuring-bfc26b72b9ae)

Reply:
I've been looking into this, but there's a few issues. 

`findCurrentUser` is taking the `userToken` from the `window` object, so it's not coming from props or state.  I'm not sure it's going to be possible to make this a selector, since it seems you can only [create a selector with state or props (https://redux.js.org/docs/recipes/ComputingDerivedData.html).
 
As for `findFacilitatorName`, I tried to do it with the [Reselect library](https://github.com/reactjs/reselect), but it ended up being really complicated to test. I'm not sure it's worth making this a selector, since we're only using it once and the memoization doesn't seem necessary as it's going to be a frequent calculation.  

I do agree with you on the value of extracting these functions, especially `findCurrentUser`, so if there's any other way I can do this I'd be happy to try. Thoughts?

Reply:
hey @laumontemayor! All good points. Don't worry about `reselect`/memoization at this point, as that's an optimization pattern we can pursue down the road. You'll likely have a nicer time defining raw selection functions that we can use in `mapStateToProps`, as those will, at the very least, allow us to dedupe logic  that appears in multiple components.

Oh! And regarding `window`! It's available globally, so you can reference `window.userToken` wherever you define the selector. And if you need `window.userToken` defined in the selector's test, you can add `window.userToken = "someStringValue"` in the `before/beforeEach` block for the test. So we're covered both in-app and in the test suite.

Reply:
(my bad, didn't mean to submit that last comment yet) Regarding `findFacilitatorName`, we're deriving the facilitator name in `LobbyStage`, but also in `RemoteRetro`. By encapsulating this in a selector, we can dedupe that logic.

Reply:
If you have any other questions about this, hit me up here, but by all means you can also ping me on slack!

Reply:
See updated PR.